on_error config for models (#9092)
fishtown-analytics/docs.getdbt.com: website/docs/docs/deploy/job-commands.md

Job command failures can mean different things for different commands. Some common reasons why a job command may fail:

> - **Failure at`dbt run`** &mdash; [`dbt run`](/reference/commands/run) executes compiled SQL model files against the current target database. It will fail if there is an error in any of the built models. Tests on upstream resources prevent downstream resources from running and a failed test will skip them. By default, if a model fails, its downstream models are also skipped. In <Constant name="core" /> v1.12+, you can set [`on_error: continue`](/reference/resource-configs/on_error) on a model to allow its downstream models to run despite the failure.

Reviewer: Hiya Bianca

"Tests on upstream resources prevent downstream resources from running and a failed test will skip them" appears to describe dbt run behavior. Since this sentence sits in the ‘Failure at dbt run’ section, it reads like it applies to dbt run, but tests are not executed by dbt run. Maybe move that sentence to the `dbt build` or `dbt test` bullets instead.

Author: Great catch! Addressed here: https://github.com/dbt-labs/docs.getdbt.com/pull/9092/commits/3a24765ec9d81aabad855d93726592ccf6f8dc0d

> - **Failure at`dbt run`** &mdash; [`dbt run`](/reference/commands/run) executes compiled SQL model files against the current target database. It will fail if there is an error in any of the built models. Tests on upstream resources prevent downstream resources from running and a failed test will skip them. By default, if a model fails, its downstream models are also skipped. In <Constant name="core" /> v1.12+, you can set [`on_error: continue`](/reference/resource-configs/on_error) on a model to allow its downstream models to run despite the failure.

Reviewer: Hiya Bianca

The [issue](https://github.com/dbt-labs/dbt-core/issues/2142) thread describes this as “letting them fail or succeed as they will” To avoid implying downstream models will succeed, consider something like “allow downstream models to still attempt to run.

Author: Addressed here: https://github.com/dbt-labs/docs.getdbt.com/pull/9092/commits/3a24765ec9d81aabad855d93726592ccf6f8dc0d

> - **Failure at`dbt run`** &mdash; [`dbt run`](/reference/commands/run) executes compiled SQL model files against the current target database. It will fail if there is an error in any of the built models. Tests on upstream resources prevent downstream resources from running and a failed test will skip them. By default, if a model fails, its downstream models are also skipped. In <Constant name="core" /> v1.12+, you can set [`on_error: continue`](/reference/resource-configs/on_error) on a model to allow its downstream models to run despite the failure.

Reviewer: ```suggestion
- **Failure at `dbt run`** &mdash; [`dbt run`](/reference/commands/run) executes compiled SQL model files against the current target database. It will fail if there is an error in any of the built models. Tests on upstream resources prevent downstream resources from running and a failed test will skip them. By default, if a model fails, its downstream models are also skipped. In <Constant name="core" /> v1.12+, you can set [`on_error: continue`](/reference/resource-configs/on_error) on a model to allow its downstream models to run despite the failure.
```

> - **Failure at `dbt test`** &mdash;  [`dbt test`](/reference/commands/test) runs tests defined on models, sources, snapshots, and seeds. A test can pass, fail, or warn depending on its [severity](/reference/resource-configs/severity). Unless you set [warnings as errors](/reference/global-configs/warnings), only an error stops the next step.

Reviewer: ```suggestion
- **Failure at `dbt test`** &mdash;  [`dbt test`](/reference/commands/test) runs tests defined on models, sources, snapshots, and seeds. A test can pass, fail, or warn depending on its [severity](/reference/resource-configs/severity). Unless you set [warnings as errors](/reference/global-configs/warnings), only an error fails the command.
```

> - **Failure at `dbt build`** &mdash; [`dbt build`](/reference/commands/build) runs models, tests, snapshots, and seeds. This command executes resources in the DAG-specified order. If any upstream resource fails, all downstream resources are skipped, and the command exits with an error code of 1. In <Constant name="core" /> v1.12+, you can set [`on_error: continue`](/reference/resource-configs/on_error) on a model to allow its downstream models to run when that model fails.

Reviewer: ```suggestion
- **Failure at `dbt build`** &mdash; [`dbt build`](/reference/commands/build) runs models, tests, snapshots, and seeds. This command executes resources in the DAG-specified order. If any upstream resource fails, all downstream resources are skipped, and the command exits with an error code of `1`. In <Constant name="core" /> v1.12+, you can set [`on_error: continue`](/reference/resource-configs/on_error) on a model to allow its downstream models to run when that model fails.
```

- **Selector failures**

>    - If a [`select`](/reference/node-selection/set-operators) matches multiple nodes and one of the nodes fails, then the job will have an exit code `1` and the subsequent command will fail. If you specified the [`—fail-fast`](/reference/global-configs/failing-fast) flag, then the first failure will stop the entire connection for any models that are in progress. 

Reviewer: Hiya @luna-bianca 

I don't know if this is in the scope of the PR but should [`—fail-fast`] be [`--fail-fast`]?

Author: ```suggestion
   - If a [`select`](/reference/node-selection/set-operators) matches multiple nodes and one of the nodes fails, then the job will have an exit code `1` and the subsequent command will fail. If you specified the [`--fail-fast`](/reference/global-configs/failing-fast) flag, then the first failure will stop the entire connection for any models that are in progress. 
```
